File: dotnet/SDL.DXA.Modules.CampaignContent/Models/TaggedProperty.cs
using Sdl.Web.Common.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SDL.DXA.Modules.CampaignContent.Models
{
    [SemanticEntity(EntityName = "TaggedProperty", Vocab = CoreVocabulary, Prefix = "e")]
    public class TaggedProperty : EntityModel
    {
        [SemanticProperty("e:name")]
        public string Name { get; set; }

        [SemanticProperty("e:value")]
        public string Value { get; set; }

        [SemanticProperty("e:target")]
        public string Target { get; set; }
    }
}

File: java/campaigncontent-dxa-module/src/main/java/com/sdl/dxa/modules/campaigncontent/model/TaggedProperty.java
package com.sdl.dxa.modules.campaigncontent.model;

import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import com.sdl.webapp.common.api.model.entity.MediaItem;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

@SemanticEntity(entityName = "TaggedProperty",  vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class TaggedProperty extends AbstractEntityModel {

    @SemanticProperty("e:name")
    private String name;

    @SemanticProperty("e:value")
    private String value;

    @SemanticProperty("e:target")
    private String target;

    @SemanticProperty("e:index")
    private Integer index;

    @SemanticProperty("e:image")
    private MediaItem image;

    @SemanticProperty("e:imageAltText")
    private String imageAltText;

    public String getName() {
        return name;
    }

    public String getValue() {
        return value;
    }

    public String getTarget() {
        return target;
    }

    public Integer getIndex() {
        return index;
    }

    public MediaItem getImage() {
        return image;
    }

    public String getImageAltText() {
        return imageAltText;
    }
}
